src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -63,18 +65,9 @@
  */
 class TreeMaker {
 
-    record CursorKey(Position position, String spelling) {
-        static CursorKey of(Cursor cursor) {
-            return new CursorKey(CursorPosition.of(cursor), cursor.spelling());
-        }
-    }
+    private final Map<Cursor.Key, Declaration> declarationCache = new HashMap<>();
 
-    private final TreeMaker parent;
-    private final Map<CursorKey, Declaration> declarationCache = new HashMap<>();
-
-    public TreeMaker(TreeMaker parent) {
-        this.parent = parent;
-    }
+    public TreeMaker() { }
```

Comment:
The logic to create "nested" tree makers is gone now - since comparison between cursors from different TUs doesn't work anyways.